app/controllers/groups_controller.rb

```diff
@@ -44,7 +51,20 @@ def update
 
   private
 
+  def authorized?
+    if params[:token]
+      authorize!
```

Comment:
I think it's a bit weird to have a `xxx?` method which raises an exception. What do you think about creating another `before_action` (i.e: `before_action :authorize_when_token!, only: [:show]`)?